js/data.js

```diff
@@ -6,7 +6,14 @@ var data = {
   nextEntryId: 1
 };
 
+var $view = document.querySelectorAll('.view');
+
 window.addEventListener('beforeunload', function (event) {
+  for (var i = 0; i < $view.length; i++) {
+    if (!$view[i].classList.value.includes('hidden')) {
+      data.view = $view[i].getAttribute('data-view');
+    }
+  }
```

Comment:
If you just update the value of this property whenever you change views for a user, you can avoid needing to do this loop every time the page unloads.